addons/point_of_sale/static/src/js/models.js

```diff
@@ -746,7 +748,11 @@ exports.PosModel = Backbone.Model.extend({
         var self = this;
         this.set('synch',{ state: 'connecting', pending: orders.length});
 
-        return self._save_to_server(orders, options).done(function (server_ids) {
+        // Need to deep copy orders to avoid deleting new orders (which are saved to "orders" in localStorage also)
+        // while previous orders are in progress of "_save_to_order" and removed them after "_save_to_order" successfull
+        var flush_orders = JSON.parse(JSON.stringify(orders));
```

Comment:
@jorenvo : after re-thinking about this, i think we should copy `orders` than writing `some mutex or some other checking` as you suggest above to respect the consistency of `orders`. 
I think `orders` should be the same before and after processing `_save_to_server`. So it helps us avoid any traps if `_save_to_server` is inherited somewhere in another module and use `orders` data to do something later.
About technically, i agree deep copy is relatively expensive operation, so i switched to shallow copy to save performance, i tested and shallow copy is enough in this case. Base on some test http://jsben.ch/#/jCucv, i chose `ES6 Object.assign()` to copy `orders`.

Waiting for your idea.
Regards